lib/rspec/mocks/error_generator.rb

```diff
@@ -291,6 +303,19 @@ def format_received_args(*args)
         args.empty? ? "(no args)" : "(" + received_arg_list(*args) + ")"
       end
 
+      def format_unexpected_with_args(*args)
+        with_args = Hash[Array(args.first).group_by { |x| x }.map { |k, v| [k, v.count] }]
```

Comment:
When I tried using the existing method, some arguments came in as `[[[:one],[:two]]]`, while others came in as `[1]` or `["wrong"]`.  `[1].first.group_by` would cause an exception, so I wrapped it up in `Array` to prevent that.   After creating a new method instead, I could have taken out the `Array` since it was always receiving consistent arguments, but I left it as is so it would still process if something unexpected came in.  

I'll re-evaluate if this will be needed after the callers get adjusted for the new method.  Might not need it anymore after that.